src/test/streams_tests.cpp

```diff
@@ -445,4 +445,29 @@ BOOST_AUTO_TEST_CASE(streams_buffered_file_rand)
     fs::remove("streams_test_tmp");
 }
 
+// This test helps compare the performance of FindByte() implementations;
+// its functionality is tested in the test above. Set the repetition
+// limit (500) to a larger value and then time the test.
+BOOST_AUTO_TEST_CASE(streams_findbyte)
+{
+    FILE* file = fsbridge::fopen("streams_test_tmp", "w+b");
+    const size_t fileSize = 200;
+    uint8_t b = 0;
+    for (size_t i = 0; i < fileSize; ++i) {
+        fwrite(&b, 1, 1, file);
+    }
+    b = 1;
+    fwrite(&b, 1, 1, file);
+    rewind(file);
+    CBufferedFile bf(file, fileSize * 2, fileSize, 0, 0);
+    for (int rep = 0; rep < 100; ++rep) {
```

Comment:
I don't fully understand how the performance increase is so significant, but why not a bench if you're worried about burdening the unit tests? I [tried](https://github.com/gzhao408/bitcoin/blob/c3f7075f538cca89f44923d9489403581650edc8/src/bench/streams_findbyte.cpp) to do this but I must be doing something wrong because I can't seem to reproduce the speedup. 😞

Reply:
The performance impact may be very compiler/architecture/stdlib dependent. I'm kind of surprised std::find has optimizations beyond the naive loop implementation in the first place on some platforms, so I certainly wouldn't be surprised if others don't have it.

Reply:
@gzhao408, thank you, I wasn't aware of `bench`. I suspect the iteration count, 100, is far too low and the difference is swamped by the noise. I increased the iteration count to 10m (`1e7`) and it showed the expected difference, master: 1,659.30 ns/op, PR: 52.66 ns/op (ratio is about 31).

I just force-pushed ([diff](https://github.com/bitcoin/bitcoin/compare/a31aa3255499ba8878ba98e4ac5a22a408c68da0..8b07e17cb2a20db8c803cfbb8491ae51a75bd450)) to remove the unit test (which was only for benchmarking, not really testing anything) and cherry-pick Gloria's benchmark. I added one more commit to increase the iteration count.